src/sql/parts/connection/connectionDialog/connectionDialogService.ts

```diff
@@ -149,16 +151,21 @@ export class ConnectionDialogService implements IConnectionDialogService {
 	}
 
 	private handleOnCancel(params: INewConnectionParams): void {
-		if (params && params.input && params.connectionType === ConnectionType.editor) {
-			this._connectionManagementService.cancelEditorConnection(params.input);
+		if (this.uiController.databaseDropdownExpanded) {
+			this.uiController.closeDatabaseDropdown();
 		} else {
-			this._connectionManagementService.cancelConnection(this._model);
-		}
-		if (params && params.input && params.input.onConnectReject) {
-			params.input.onConnectReject();
+			if (params && params.input && params.connectionType === ConnectionType.editor) {
+				this._connectionManagementService.cancelEditorConnection(params.input);
+			} else {
+				this._connectionManagementService.cancelConnection(this._model);
+			}
+			if (params && params.input && params.input.onConnectReject) {
+				params.input.onConnectReject();
+			}
+			this._connectionDialog.resetConnection();
+			this._connecting = false;
 		}
-		this._connectionDialog.resetConnection();
-		this._connecting = false;
+		this.uiController.databaseDropdownExpanded = false;
```

Comment:
nit: I would put this inside the `if` block on line 154 since it will already be false in the `else` clause